app/components/Button.tsx

```diff
@@ -1,6 +1,7 @@
 import { Icon } from '~/types/Icon';
 
 interface ButtonProps {
+  type?: 'submit' | 'reset';
```

Comment:
Can remove optional if you have give default value to the prop

Reply:
at some places we might not need to pass the type prop, so i think we should keep it optional, but if the user passes a type prop we can keep it empty instead of having a default value then ?